### ISSUE TITLE
Registration Flow

### ISSUE THREAD

**Issue by pingskills** [cubing/worldcubeassociation.org#758]
When a user has preferred events, then registers for those events, and then subsequently updates their registration to delete one of those preferred events, the form still shows all the preferred events selected. The database is correct but to the user it looks like they are still registered for all their preferred events. Once a registration has been created we should no longer be using the preferred events. This should only be shown for a new registration.

**Comment by jonatanklosko** [comment]
@pingskills I cannot reproduce it, am I missing any step:
- set sepreferred events: 3x3x3 and 4x4x4
- go to competion's page
- click Register
- send form clicking Register!
- uncheck 4x4x4
- click Update registration

**Comment by pingskills** [comment]
@jonatanklosko ah, it only happens if you uncheck all events. Which is not a very likely scenario.

**Comment by jonatanklosko** [comment]
Ok I investigated it and that's a bit tricky.
Basically the problem is caused by our unworkable regitration events count validation, the problem is well described [here in the *Validating the count of nested attributes* section](http://homeonrails.com/2012/10/validating-nested-associations-in-rails/). It's easy to fix.
Apart from this I was surprised seeing my preferred events checked on the registration *edit* page (because I implemented this only for the *register* one) but then I realised that user is allowed both to create and change his registration on the *register* page/action, whereas the *edit* page is meant for organizators.
Summary fix is pretty straightforward (the validation).
I'm having a competition this weekend (and I'm travelling now) so I won't be able to fix this until Monday evening. Feel free to create PR with the fix if you have time =)
